readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -347,7 +360,9 @@ class PdfBuilder(BaseSphinx):
 
     def build(self):
         self.clean()
-        cwd = self.project.conf_dir(self.version.slug)
+        config_file = self.config.sphinx.configuration
+        config_dir = os.path.dirname(config_file) if config_file else None
+        cwd = config_dir or self.project.conf_dir(self.version.slug)
```

Comment:
This logic looks duplicated, is there a way to reduce this duplication if so?

Reply:
I knew you were not going to like it p: haha, so all this is to make sure that we are keeping the previous behavior. If a config file doesn't exist, we create one, and also we try to find one in